CRM/Core/Component.php

```diff
@@ -27,7 +27,7 @@ class CRM_Core_Component {
   /**
    * @var array
    */
-  public static $_contactSubTypes = NULL;
+  public static $_contactSubTypes;
 
```

Comment:
There seems to be a fashion for removing `= NULL`. As a reader, I like having a reference to null. For me, these three formulations are equally effective at communicating the same intention:

```php
/**
 * @var array
 */
protected $foo = NULL;

/**
 * @var array|null
 */
protected $foo;

protected ?array $foo;
```

These notations reassure me that the author and I and other readers (*who come from different backgrounds with different languages/frameworks*) will have a common understanding that *it is normal+expected for this field to have a period of NULLity*. This influences how you organize guards/references/etc. For folks with a lot of experience in this repo, maybe you don't need the hint ("This is a `static` underscore property in a `CRM_*` class, so it's probably intended to be `null` until someone calls a lazy-loading getter")... but... it really doesn't hurt you to have it...

(Another tangent... it's remarkably silly that Civi's linter requires `NULL` for PHP code and `null` for PHP docblocks... )

Reply:
Since this turns out to be not needed I just removed it - however I think the
```
/**
 * Contact Subtypes.
 *
 * @var array|null
 */
```
syntax looks like it does what you wanted and complies with a popular static code analyser (https://github.com/kalessil/phpinspectionsea - which has over 1 million downloads within php storm). I do base my style changes on external standards - mostly a combo of the full drupal standard and the various things that seem to be agreed best practices in phpstorm world (since those obviously reflect a largish community and are easily implemented)




Note it is not civi's linter that sets the NULL vs null - it's the drupal standard that we have been implementing (although it's possible that it has changed since we forked a while ago)

Reply:
Forked off a separate issue about the null/NULL conflict: https://www.drupal.org/project/coder/issues/3259288

Reply:
Ok - I think I'd rather switch to the psr2 than try to change the drupal standard - I have a feeling it wouldn't be insanely hard to switch over - but in general I'm more interesting in complying with an external standard than trying to come up with our own and I don't care if that standard isn't 100% to my taste